### ISSUE TITLE
I cant no longer send messages

### ISSUE THREAD

**Issue by Martial83** [AsamK/signal-cli#609]
Hi, 
I can receive messages, but error when i try to send one.
Can somebody help me ?

 java -version
openjdk version "11.0.9.1" 2020-11-04
OpenJDK Runtime Environment (build 11.0.9.1+1-post-Raspbian-1deb10u2)
OpenJDK Server VM (build 11.0.9.1+1-post-Raspbian-1deb10u2, mixed mode)

signal-cli --verbose -u +xxxxxxx send -m "tagada" +xxxxxxx
2021-05-02T18:03:07.484+0200 [main] WARN LibSignal - [PhoneNumberFormatter]: Got local CC: FR
2021-05-02T18:03:07.503+0200 [main] INFO LibSignal - [WebSocketConnection]: connect()
2021-05-02T18:03:07.534+0200 [main] INFO LibSignal - [WebSocketConnection]: connect()
2021-05-02T18:03:07.971+0200 [main] WARN LibSignal - [SignalServiceMessageSender]: No attachments present...
2021-05-02T18:03:08.300+0200 [main] WARN LibSignal - [SignalServiceMessageSender]: java.io.IOException: No connection!
        at org.whispersystems.signalservice.internal.websocket.WebSocketConnection.sendRequest(WebSocketConnection.java:206)
        at org.whispersystems.signalservice.api.SignalServiceMessagePipe.send(SignalServiceMessagePipe.java:194)
        at org.whispersystems.signalservice.api.SignalServiceMessageSender.sendMessage(SignalServiceMessageSender.java:1545)
        at org.whispersystems.signalservice.api.SignalServiceMessageSender.sendMessage(SignalServiceMessageSender.java:316)
        at org.asamk.signal.manager.Manager.sendMessage(Manager.java:1393)
        at org.asamk.signal.manager.Manager.sendMessage(Manager.java:1319)
        at org.asamk.signal.manager.Manager.sendMessage(Manager.java:982)
        at org.asamk.signal.dbus.DbusSignalImpl.sendMessage(DbusSignalImpl.java:94)
        at org.asamk.signal.commands.SendCommand.handleCommand(SendCommand.java:132)
        at org.asamk.signal.commands.DbusCommand.handleCommand(DbusCommand.java:15)
        at org.asamk.signal.App.handleLocalCommand(App.java:204)
        at org.asamk.signal.App.init(App.java:164)
        at org.asamk.signal.Main.main(Main.java:49)

2021-05-02T18:03:08.307+0200 [main] WARN LibSignal - [SignalServiceMessageSender]: [sendMessage] Pipe failed, falling back...
2021-05-02T18:03:08.367+0200 [OkHttp https://textsecure-service.whispersystems.org/...] INFO LibSignal - [WebSocketConnection]: onOpen() connected
2021-05-02T18:03:08.367+0200 [OkHttp https://textsecure-service.whispersystems.org/...] INFO LibSignal - [WebSocketConnection]: onOpen() connected
1619971387499
2021-05-02T18:03:09.259+0200 [main] INFO LibSignal - [WebSocketConnection]: disconnect()
2021-05-02T18:03:09.265+0200 [main] INFO LibSignal - [WebSocketConnection]: disconnect()

**Comment by AsamK** [comment]
The message has been sent successfully according to your output.
The stack trace is just a warning.
The command printed the timestamp of the sent message so it should have been transmitted to the server: `1619971387499`

**Comment by Martial83** [comment]
Thank @AsamK but i don't receive it on the receiver.

**Comment by dreem2001** [comment]
The same here. The messages have never been delivered to recipient.

**Comment by AsamK** [comment]
Maybe the session state got out of sync and the recipient fails to decrypt the message.
Does resetting the session resolve the issue?
`signal-cli -u +xxxxxxx send -e +xxxxxxx`

**Comment by Martial83** [comment]
No, that does not solve the problem

**Comment by Martial83** [comment]
No other idea on the origin of the problem?

**Comment by dreem2001** [comment]
Any ideas will be much appreciated. Reregistering doesn't help as well. I do not receive any message for a week. Thanks a in advance.

**Comment by AsamK** [comment]
Can you get a debug log from the recipient? Either signal-cli verbose output or Signal-Android debug log.
Without that I don't have a clue what's going wrong. signal-cli printed the message timestamp which means the message has been delivered to the server and for some reason it doesn't arrive at the recipient or is rejected by it for some reason...

**Comment by dreem2001** [comment]
@AsamK  I registered a new ID and sent test from old (broken) ID to the new ID. 
signal-cli receive output:
```2021-05-06T00:12:45.438+0300 [main] INFO LibSignal - [WebSocketConnection]: connect()
2021-05-06T00:12:46.621+0300 [OkHttp https://textsecure-service.whispersystems.org/...] INFO LibSignal - [WebSocketConnection]: onOpen() connected
Envelope from: unknown source
Timestamp: 1620249155548 (2021-05-05T21:12:35.548Z)
Sent by unidentified/sealed sender
Sender: +7xxxxxxx4587 (device: 1)
Server timestamps: received: 1620249157409 (2021-05-05T21:12:37.409Z) delivered: 1620249166664 (2021-05-05T21:12:46.664Z)
Message timestamp: 1620249155548 (2021-05-05T21:12:35.548Z)
Body: test12
Profile key update, key length: 32

2021-05-06T00:12:53.576+0300 [main] INFO LibSignal - [WebSocketConnection]: disconnect()

**Comment by Martial83** [comment]
> 
> 
> Can you get a debug log from the recipient? Either signal-cli verbose output or Signal-Android debug log.
> Without that I don't have a clue what's going wrong. signal-cli printed the message timestamp which means the message has been delivered to the server and for some reason it doesn't arrive at the recipient or is rejected by it for some reason...

What can i give you?
The recipient is successfully receiving messages from others but nothing from this

**Comment by dreem2001** [comment]
Well. Something happened on Signal side. I managed to have it work.
1) It's necessary to mark the sender as verified. (In case of re-register - reset security code as well)
2) It's necessary to send from recipient a message and receive it on signal-cli side

Only after two these steps it started working. 

@AsamK  any ideas why only I became a loser?

**Comment by Martial83** [comment]
I have done that, but no chance, and, now, error on receive:
Envelope from: unknown source
Timestamp: 1620288753499 (2021-05-06T08:12:33.499Z)
Sent by unidentified/sealed sender

What i have done:
pi@raspberrypi:/var/log $signal-cli -u +3311111111 register --voice
Captcha required for verification, use --captcha CAPTCHA
To get the token, go to https://signalcaptchas.org/registration/generate.html
Check the developer tools (F12) console for a failed redirect to signalcaptcha://
Everything after signalcaptcha:// is the captcha token.
pi@raspberrypi:/var/log $ signal-cli -u +3311111111 register --captcha "03AGd.......qqETEzp3DABWLmxVw"
signal-cli -u +33781830742 verify 964440

**Comment by dreem2001** [comment]
In my case it still works. But only for recipients where I marked ID as verified and send\receive messages. 
So did you send message from recipient to signalcli's ID and receive it there and then send back a message to the recipient from cli? I believe this is the key point to let signal.org a sign that it's alive ID sending\receiving messages. 
Once I did it I have not seen "unknown\stolen" in verbose report from receive command of signal-cli and messages go to recipients.
